internal/operands/tekton-tasks/tekton-tasks.go

```diff
@@ -78,76 +76,74 @@ type tektonTasks struct {
 
 var _ operands.Operand = &tektonTasks{}
 
-func New(bundle *tektonbundle.Bundle) *tektonTasks {
-	tt := &tektonTasks{
+func New(bundle *tektonbundle.Bundle) operands.Operand {
+	newTasks := []pipeline.Task{}
+	for _, task := range bundle.Tasks {
+		if _, ok := AllowedTasks[task.Name]; ok {
+			newTasks = append(newTasks, task)
+		}
+	}
+	bundle.Tasks = newTasks
+
+	newServiceAccounts := []v1.ServiceAccount{}
+	for _, serviceAccount := range bundle.ServiceAccounts {
+		if _, ok := AllowedTasks[strings.TrimSuffix(serviceAccount.Name, "-task")]; ok {
+			newServiceAccounts = append(newServiceAccounts, serviceAccount)
+		}
+	}
+	bundle.ServiceAccounts = newServiceAccounts
+
+	newRoleBinding := []rbac.RoleBinding{}
+	for _, roleBinding := range bundle.RoleBindings {
+		if _, ok := AllowedTasks[strings.TrimSuffix(roleBinding.Name, "-task")]; ok {
+			newRoleBinding = append(newRoleBinding, roleBinding)
+		}
+	}
+	bundle.RoleBindings = newRoleBinding
+
+	newClusterRole := []rbac.ClusterRole{}
+	for _, clusterRole := range bundle.ClusterRoles {
+		if _, ok := AllowedTasks[strings.TrimSuffix(clusterRole.Name, "-task")]; ok {
+			newClusterRole = append(newClusterRole, clusterRole)
+		}
+	}
+	bundle.ClusterRoles = newClusterRole
+
+	return &tektonTasks{
```

Comment:
**future PR** - Can you pass the local variables here, `newTasks`, `newServiceAccounts`, etc...